backend/app/services/workflow_engines/base_engine.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod, abstractstaticmethod
+
+from app.helpers import WorkflowState
+from app.models.experiment import Experiment
+from app.models.experiment_run import ExperimentRun
+
+
+class WorkflowConnectionExcpetion(Exception):
+    pass
+
+
+class WorkflowBaseEngine(ABC):
```

Comment:
I would rename the class to `WorkflowEngineBase and the whole module simply to `base`